### DISCUSSION TITLE
Will RPC be supported in flatcc?

### DISCUSSION THREAD

**Discussion by kojung** [dvidelabs/flatcc#191] [Q&A] [answered]
First of all, thanks for the very nice flatbuffer compiler/runtime written in C. I was able to build the library in an embedded processor without any issues.
I see that RPC is currently parsed but ignored during code generation steps. Are there plans to support RPC code generation in the future? If so, what is the rough timeline?
Thanks.

**Comment by mikkelfj** [comment, marked as answer]
Thanks,

there are no current plans for RPC and especially not gRPC because it is very tool heavy.
At some point I could imagine adding RPC for MQTT 5.0 RPC calls, but there are no current plans, and I am not really sure what RPC gives over just using tables in message processors - what would RPC logic add?

However, RPC was recently added to the binary bfbs schema, so it should be relatively easy to read a bfbs file and generate custom RPC logic as needed. This is only master branch but will be included in the next release (no timeline, but here is more than enough for new release already).

**Reply by kojung** [reply]
Thanks for the quick and helpful reply. I agree that RPC logic does not add too much. I just wanted to avoid writing new code until knowing the overall project roadmap.
